hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerServiceTest.java

```diff
@@ -164,6 +165,35 @@ public void createThing() {
     }
 
     @Test
+    @Description("Tests the target attribute update by calling the same method that incoming RabbitMQ messages would access.")
+    public void updateAttributes() {
+        final String knownThingId = "1";
+        final MessageProperties messageProperties = createMessageProperties(MessageType.EVENT);
+        messageProperties.setHeader(MessageHeaderKey.THING_ID, "1");
+        messageProperties.setHeader(MessageHeaderKey.TOPIC, "UPDATE_ATTRIBUTES");
+        final AttributeUpdate attributeUpdate = new AttributeUpdate();
+        attributeUpdate.getAttributes().put("testKey1", "testValue1");
+        attributeUpdate.getAttributes().put("testKey2", "testValue2");
+
+        final Message message = amqpMessageHandlerService.getMessageConverter().toMessage(attributeUpdate,
+                messageProperties);
+
+        final ArgumentCaptor<String> targetIdCaptor = ArgumentCaptor.forClass(String.class);
+        final ArgumentCaptor<Map> attributesCaptor = ArgumentCaptor.forClass(Map.class);
```

Comment:
If you use @Captor private ArgumentCaptor<Map<String, String>> attributesCaptor; you can avoid the nested generic problem.